src/signal.rs

```diff
@@ -1483,3 +1484,115 @@ mod calendar {
+            unsafe {
+                let f: Box<Box<Fn(StatusIcon, &EventButton) -> bool + 'static>> = Box::new(Box::new(f));
+                connect(self.to_glib_none().0 as *mut _, "button-release-event",
+                    transmute(event_trampoline), into_raw(f) as *mut _)
+            }
+        }
+
+        fn connect_popup_menu<F: Fn(StatusIcon, u32, u32) + 'static>(&self, f: F) -> u64 {
+            unsafe {
+                let f: Box<Box<Fn(StatusIcon, u32, u32) + 'static>> = Box::new(Box::new(f));
+                connect(self.to_glib_none().0 as *mut _, "popup-menu",
+                    transmute(popup_menu_trampoline), into_raw(f) as *mut _)
+            }
+        }
+
+        fn connect_query_tooltip<F: Fn(StatusIcon, i32, i32, bool, Tooltip) -> bool + 'static>(&self, f: F) -> u64 {
+            unsafe {
+                let f: Box<Box<Fn(StatusIcon, i32, i32, bool, Tooltip) -> bool + 'static>> = Box::new(Box::new(f));
+                connect(self.to_glib_none().0 as *mut _, "query-tooltip",
+                    transmute(query_tooltip_trampoline), into_raw(f) as *mut _)
+            }
+        }
+
+        fn connect_scroll_event<F: Fn(StatusIcon, &EventScroll) -> bool + 'static>(&self, f: F) -> u64 {
+            unsafe {
+                let f: Box<Box<Fn(StatusIcon, &EventScroll) -> bool + 'static>> = Box::new(Box::new(f));
+                connect(self.to_glib_none().0 as *mut _, "scroll-event",
+                    transmute(event_trampoline), into_raw(f) as *mut _)
+            }
+        }
+
+        fn connect_size_changed<F: Fn(StatusIcon, i32) -> bool + 'static>(&self, f: F) -> u64 {
+            unsafe {
+                let f: Box<Box<Fn(StatusIcon, i32) -> bool + 'static>> = Box::new(Box::new(f));
+                connect(self.to_glib_none().0 as *mut _, "size-changed",
+                    transmute(size_changed_trampoline), into_raw(f) as *mut _)
+            }
+        }
+    }
+
+    extern "C" fn void_trampoline(this: *mut GtkStatusIcon, f: &Box<Fn(StatusIcon) + 'static>) {
+        unsafe {
+            f(from_glib_none(this));
+        }
+    }
+
+    extern "C" fn event_trampoline(this: *mut GtkStatusIcon, event: *mut EventButton, f: &Box<Fn(StatusIcon, &EventScroll) -> bool + 'static>) -> gboolean {
+        unsafe {
+            f(from_glib_full(this), transmute(event)).to_glib()
+        }
+    }
+
+    extern "C" fn popup_menu_trampoline(this: *mut GtkStatusIcon, button: c_uint, activate_time: c_uint, f: &Box<Fn(StatusIcon, u32, u32) + 'static>) {
+        unsafe {
+            let icon: StatusIcon = from_glib_none(this);
+            f(icon, button, activate_time);
+        }
+    }
+
+    extern "C" fn query_tooltip_trampoline(this: *mut GtkStatusIcon, x: c_int, y: c_int, keyboard_mode: bool, _tooltip: *mut GtkTooltip, f: &Box<Fn(StatusIcon, i32, i32, bool, Tooltip) -> bool + 'static>) -> gboolean {
```

Comment:
`keyboard_mode` here should be `gboolean`. C.f. https://developer.gnome.org/gtk3/stable/GtkStatusIcon.html#GtkStatusIcon-query-tooltip